### ISSUE TITLE
Unable to build for Python 3.13 on Linux

### ISSUE THREAD

**Issue by pjyan** [ludwigschwardt/python-gnureadline#70]
python 3.13.0a3
gnureadline-8.1.2

As mentioned in https://github.com/python/cpython/issues/117669, _PyOS_ReadlineTState was removed from cpython 3.13

The error is
```shell
      Modules/3.x/clinic/readline.c.h:34:10: warning: implicit declaration of function ‘_PyArg_CheckPositional’; did you mean ‘PyErr_CheckSignals’? [-Wimplicit-function-declaration]
           if (!_PyArg_CheckPositional("read_init_file", nargs, 0, 1)) {
                ^~~~~~~~~~~~~~~~~~~~~~
                PyErr_CheckSignals
      In file included from Modules/3.x/readline.c:943:0:
      Modules/3.x/clinic/readline.c.h: In function ‘readline_append_history_file’:
      Modules/3.x/clinic/readline.c.h:143:17: warning: implicit declaration of function ‘_PyLong_AsInt’; did you mean ‘PyLong_AsInt’? [-Wimplicit-function-declaration]
           nelements = _PyLong_AsInt(args[0]);
                       ^~~~~~~~~~~~~
                       PyLong_AsInt
      Modules/3.x/clinic/readline.c.h: In function ‘readline_replace_history_item’:
      Modules/3.x/clinic/readline.c.h:441:9: warning: implicit declaration of function ‘_PyArg_BadArgument’; did you mean ‘PyErr_BadArgument’? [-Wimplicit-function-declaration]
               _PyArg_BadArgument("replace_history_item", "argument 2", "str", args[1]);
               ^~~~~~~~~~~~~~~~~~
               PyErr_BadArgument
      Modules/3.x/readline.c: In function ‘readline_until_enter_or_signal’:
      Modules/3.x/readline.c:1356:34: error: ‘_PyOS_ReadlineTState’ undeclared (first use in this function); did you mean ‘PyOS_Readline’?
                   PyEval_RestoreThread(_PyOS_ReadlineTState);
                                        ^~~~~~~~~~~~~~~~~~~~
                                        PyOS_Readline
      Modules/3.x/readline.c:1356:34: note: each undeclared identifier is reported only once for each function it appears in
      Modules/3.x/readline.c: In function ‘call_readline’:
      Modules/3.x/readline.c:1387:5: warning: implicit declaration of function ‘_Py_SetLocaleFromEnv’ [-Wimplicit-function-declaration]
           _Py_SetLocaleFromEnv(LC_CTYPE);
           ^~~~~~~~~~~~~~~~~~~~
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
```

**Comment by ludwigschwardt** [comment]
Hi, I'm currently busy making a new release that fixes issues with 3.12 and I decided to have a go at 3.13 as well. Unfortunately it's a bit too early... Tools like tox and pyenv are not quite up to speed yet, and there are many internal changes besides `_PyOS_ReadlineTState`. I'll wait until at least a beta release of 3.13 before I try again.

**Comment by pjyan** [comment]
> Hi, I'm currently busy making a new release that fixes issues with 3.12 and I decided to have a go at 3.13 as well. Unfortunately it's a bit too early... Tools like tox and pyenv are not quite up to speed yet, and there are many internal changes besides `_PyOS_ReadlineTState`. I'll wait until at least a beta release of 3.13 before I try again.

Thanks in advance. I guess there are not a lot of requirements for gnureadline based on 3.13 at this time...
